source/physics/IonizationClustering.cc

```diff
@@ -109,15 +109,13 @@ namespace nexus {
     // and N is the average number of charges.
 
     // Fetch the W_i and F from the material properties table
-    //G4MaterialPropertiesTable* mpt =
-    //  track.GetMaterial()->GetMaterialPropertiesTable();
-    //if (!mpt)
-    //  return G4VRestDiscreteProcess::PostStepDoIt(track, step);
-
-    //G4double ioni_energy = mpt->GetConstProperty("IONIZATIONENERGY");
-    //G4double fano_factor = mpt->GetConstProperty("FANOFACTOR");
-    G4double ioni_energy = 22.4 * eV;
-    G4double fano_factor = .15;
+    G4MaterialPropertiesTable* mpt =
+      track.GetMaterial()->GetMaterialPropertiesTable();
+    if (!mpt)
+      return G4VRestDiscreteProcess::PostStepDoIt(track, step);
```

Comment:
Shouldn't this be an error?

Reply:
Yes, it would be probably better.

Reply:
Also can it be done at class initialization instead of at every call to `PostStepDoIt`?

Reply:
You can't do it in the class initialization, because you need access to the track to know in which material you are passing through at each moment.